firewall/privacy_mapper.go

```diff
@@ -812,6 +808,89 @@ func handleUpdatePolicyResponse(db firewalldb.PrivacyMapDB,
 	}
 }
 
+func handleWalletBalanceResponse(_ firewalldb.PrivacyMapDB,
+	flags session.PrivacyFlags,
+	randIntn func(int) (int, error)) func(ctx context.Context,
+	r *lnrpc.WalletBalanceResponse) (proto.Message, error) {
+
+	return func(_ context.Context, r *lnrpc.WalletBalanceResponse) (
+		proto.Message, error) {
+
+		totalBalance, err := maybeHideAmount(
+			flags, randIntn, r.TotalBalance,
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		confirmedBalance, err := maybeHideAmount(
+			flags, randIntn, r.ConfirmedBalance,
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		unconfirmedBalance, err := maybeHideAmount(
+			flags, randIntn, r.UnconfirmedBalance,
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		lockedBalance, err := maybeHideAmount(
+			flags, randIntn, r.LockedBalance,
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		reservedBalanceAnchorChan, err := maybeHideAmount(
+			flags, randIntn, r.ReservedBalanceAnchorChan,
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		accountBalance := r.AccountBalance
+		if !flags.Contains(session.ClearAccountBalances) {
```

Comment:
in what scenario would the "clear account balances" be useful?

Reply:
or rather: why is that separate from "ClearAmounts"?

Reply:
just checking for understanding: do we ever expect to have the WalletBalances API available for a feature where the clear amounts flag is not set?

Reply:
I removed `ClearAccountBalances` as it is too specific and changed it to obfuscate the amounts instead.

> just checking for understanding: do we ever expect to have the WalletBalances API available for a feature where the clear amounts flag is not set?

I'm not sure about that, it could maybe be useful for liquidity management, but hard to say